### DISCUSSION TITLE
Replacing hiredis with libvalkey?

### DISCUSSION THREAD

**Discussion by mkmkme** [ccache/ccache#1480] [Development]
Hello! Did you consider replacing `hiredis` with `libvalkey` due to [Redis license change](https://azure.microsoft.com/en-us/blog/redis-license-update-what-you-need-to-know/)? `libvalkey` is a fork of `hiredis` that is being maintained by the community.

If you are interested, I could prepare a pull request with corresponding changes.

**Comment by afbjorklund** [comment]
I don't think there has been a license change to libhiredis?

It already works with different servers, beyond just redis

**Comment by afbjorklund** [comment]
This also goes for the testing scripts, which you can run with valkey instead of with redis. The current programs need to exist, for compatibility reasons.

> For compatibility with Redis, we create symlinks from the Redis names (`redis-server`, `redis-cli`, etc.) to the Valkey binaries installed by `make install`
